.github/workflows/codeql.yml

```
@@ -0,0 +1,78 @@
+
+# CodeQL over the shipped library, complementing the clang-tidy, cppcheck-MISRA,
+# sanitizer and coverage gates in ci.yml. Findings land in Security -> Code
+# scanning rather than the job log, and the lane is advisory, not required.
+#
+# The build decides what is analysed, so this lane builds ci/consumer-smoke —
+# the documented FetchContent adoption, the same steps the published example
+# uses — rather than a bespoke configuration.
+#
+# Rationale, scope and triage convention: docs/ci.md, "Code scanning".
+
+on:
```

YAMLlint (1.37.1)

[warning] truthy value should be one of [false, true]

(truthy)